BEIMA.Client/services/ApiClient.js

```diff
@@ -0,0 +1,32 @@
+//pull in the vars file to get the API URL
+require('dotenv').config({ path: '../.env' })
+
+//axios package includes it's own try catch and error output functions 
+
+const axios = require('axios');
+
+const API_URL = process.env.REACT_APP_API_URL;
+
+/// <summary>
+/// Gets a device with the specified ID
+/// </summary>
+/// <param name="deviceId">The device ID to retrieve from the database</param>
+/// <returns>JSON object containing the HTTP status code and the API response</returns>
+const GetDevice = async (deviceId) => {
+  //performs the get and returns the data or error
+  const response = await axios.get(API_URL + "device/" + deviceId).then(response => { 
+      let stringdata = JSON.stringify(response.data);
+      return JSON.parse('{ "status" : "' + response.status + '", "response" : ' + JSON.stringify(response.data) + '}');
+    }).catch(function (error) {
+        if (error.response) {
+        return JSON.parse('{ "status" : "' + error.response.status + '", "response" : ' + JSON.stringify(error.response.data) + '}');
+      }
+  });
+  
+  return response;
+}
+
+//module.exports.<Custom Name> = <Function Name>
+//OR
+//module.exports.<Function Name> = <Function Name>
+module.exports.GetDevice = GetDevice;
```

Comment:
I think `export default GetDevice` will work here too. That's what we've been following for the other JS code.

Reply:
Would that interfere with other functions being included in the file?

Reply:
There's currently only one function in this file, and it might be a good idea to keep each API call in its own file, like how the backend is doing. But if we want to include all the functions in one file, then you can add `export default` in front of the function declaration

Reply:
You can just put export infront of const GetDevice... if you are wanting to have multiple exported functions in the file. I think the module.exports you have also works just as well.

Reply:
I tried a few different ways of exporting and a few complained when trying to call them externally, I'll try the export in front of the function name and see how it likes it, I can't remember if I tried that or not.

Reply:
One per file would limit the amount of external code that gets loaded per page, and the "show one device" doesn't need to know about device/device type list functionality and vice versa (or any other operation).

I'll swap it tonight